certbot/CHANGELOG.md

```diff
@@ -24,6 +24,10 @@ More details about these changes can be found on our GitHub repo.
 
 * Added the ability to remove email and phone contact information from an account 
   using `update_account --register-unsafely-without-email`
+* CLI flag `--key-type` has been added to specify 'rsa' or 'ecdsa' (default 'rsa').
+  Only accepts a single value at this time.
+* CLI flag `--elliptic-curve` has been added which takes an NIST/SECG elliptic curve,
```

Comment:
This needs to be made more precise. From what I read, only `secp256k1`, `secp384r1` and `secp521r1` are supported by the different browser vendors (probably defined in a standard).

Reply:
> probably defined in a standard

The CA/Browser Forum Baseline Requirements to be exact. From "6.1.5 Key sizes":

> For ECDSA key pairs, the CA SHALL:
> •  Ensure that the key represents a valid point on the NIST P-256, NIST P-384 or NIST P-521 elliptic curve.

The naming of the curves is a little bit difficult though. For example, for every key size there are two curves: one with parameters associated with a Koblitz curve (denoted by the `k` in the curve name) and one with verifiably random parameters (denoted by the `r` in the curve name). As we can see from the [SEC 2: Recommended Elliptic Curve Domain Parameters (Version 2.0)](https://www.secg.org/sec2-v2.pdf) Table 2, the NIST recommended curves are the **random** curves. I.e., with a `r` in the `secp` cuve name. Unfortunately, you're not going to find the random parameter key size 256 by its "secp" name in OpenSSL. As we can read in [RFC 5480](https://tools.ietf.org/html/rfc5480#section-2.1.1.1), for some reason the `secp256r1` curve is referred to as `prime256v1` (also in OpenSSL and in the [Current Mozilla Security/Server Side TLS Wiki document](https://wiki.mozilla.org/index.php?title=Security/Server_Side_TLS&oldid=1229478#Intermediate_compatibility_.28recommended.29).

TL;DR: sed 's/secp256k1/secp256r1/' :wink:

Reply:
Ah. Thanks :) I think I replaced it consistently, and also noticed I had `secp521k1` somewhere in my changes.

Reply:
Nothing to add. Indeed these curves are the recommended ones, and the overall design we agreed for ECDSA keys in Certbot was to follow Mozilla recommandations here.